src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -220,7 +246,7 @@ class FileProcessor(
   private def createMetadataJson(metadata: List[BagitMetadataObject]): IO[String] = {
     Stream
       .emit[IO, List[BagitMetadataObject]](metadata)
-      .through(_.map(bagitMetadataObject => bagitMetadataObject.asJson.printWith(Printer.noSpaces)))
+      .through(_.map(bagitMetadataObjects => bagitMetadataObjects.asJson.printWith(Printer.noSpaces)))
```

Comment:
I know I wrote this one but the `Stream.emit...` isn't necessary. You can just do 
```scala
uploadAsFile(metadata.asJson.printWith(Printer.noSpaces), "metadata.json")
```
I don't know why I did this. It's not like it saves memory because the metadata object needs to be stored before this happens.